repos/system_upgrade/common/libraries/rpms.py

```diff
@@ -1,7 +1,45 @@
 from leapp.libraries import stdlib
+from leapp.libraries.common.config.version import get_source_major_version
 from leapp.models import InstalledRPM
 
 
+class LeappComponents(object):
+    """
+    Supported component values to be used with get_packages_function:
+    * FRAMEWORK - the leapp project
+    * REPOSITORY - the leapp-repository project
+    * COCKPIT - the cockpit-leapp project
+    * TOOLS - miscellaneous tooling like snactor
+    """
+    FRAMEWORK = 'framework'
+    REPOSITORY = 'repository'
+    COCKPIT = 'cockpit'
+    TOOLS = 'tools'
```

Comment:
The docstring is incomplete. It still does not clarify correctly what to expect mainly in framework and tools components. The framework should clearly specify it does not include snactor (which is part of the framework project) and it includes also the leapp tool. Hence, the tools component does not contain the leapp tool.

Reply:
```suggestion
    """
    Supported component values to be used with get_packages_function:
    * FRAMEWORK - the core of the leapp project: the leapp executable and
      associated leapp libraries
    * REPOSITORY - the leapp-repository project
    * COCKPIT - the cockpit-leapp project
    * TOOLS - miscellaneous tooling like snactor
    """
    FRAMEWORK = 'framework'
    REPOSITORY = 'repository'
    COCKPIT = 'cockpit'
    TOOLS = 'tools'
```